src/Multisig.vy

```diff
@@ -4,7 +4,7 @@
 EIP712DOMAINTYPE_HASH: constant(bytes32) = keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract,bytes32 salt)")
 NAME_HASH:             constant(bytes32) = keccak256("tiny multisig")
 VERSION_HASH:          constant(bytes32) = keccak256("1")
-TXTYPE_HASH:           constant(bytes32) = keccak256("MultiSigTransaction(address destination,uint256 value,bytes data,uint256 nonce)")
+TXTYPE_HASH:           constant(bytes32) = keccak256("MultiSigTransaction(address destination,uint256 value,bytes data,uint256 nonce, uint256 expiry)")
```

Comment:
mind the extra space, this is technically wrong encoding

Reply:
also we have removed the nonce, so 1) remove nonce 2) mind the extra space

Reply:
@nmushegian I think this current `nonce` held in state + expiry stored in signature is probably a good approach for a this. 
- Replays are prevented in the most straightforward manner
- Expiration allows reasonable window to re-attempt reverted tx
- No risk of bricking
- Unexpired txs that want to be invalidated immediately can always be voided with a no-op tx  that increases the nonc

Other ways such as storing the last action time in state instead of nonce
- Get convoluted to prevent replays within same block
- Require special cases around using `expiry == 0` or other edges to avoid bricking with a bad expiry time
- Invalidating a previously failed but unexpired transaction requires more special thinking around what expiry to use

It could be done, but it wouldn't be a simple multisig anymore.